### DISCUSSION TITLE
png to avif results in weird stains on the image

### DISCUSSION THREAD

**Discussion by ElieSangeunLee** [ImageMagick/ImageMagick#5248] [Help]
Image magick version: 
```
> magick --version
Version: ImageMagick 7.1.0-37 Q16-HDRI x86_64 20104 https://imagemagick.org
```

OS: mac OS 10.15.7

Description:
I want to convert a PNG image to AVIF, but after converting, the result image has a sort of red stains here and there.

![originalcrop](https://user-images.githubusercontent.com/73095294/174551793-3399d940-0632-4a93-8a71-99948a0ef031.png)
1500*1500 png
Above is the original image(cropped)

![imagemagickconversion](https://user-images.githubusercontent.com/73095294/174551824-10d66cd8-32b3-43e0-9d95-c4a684342a04.png)
750*750 avif
Above is the image converted through Imagemagick and it has stains. The commands I tried are following:
```
magick original.png -quality 90 -resize 750 converted.avif
magick original.png -quality 90 -scale 50% converted.avif
magick original.png -quality 90 -thumbnail 750 converted.avif
```

![onlineconversion](https://user-images.githubusercontent.com/73095294/174551820-3a4c0057-2b71-46bd-85e8-3b0aa6d3a0d6.png)
750*750 avif
And this is the image from online conversion site and it is exactly what i expected, just resizing without any stains.

Could you help me find out what commands I have to use to get the proper result without any stains?

**Comment by fmw42** [comment]
What do you get from 

magick -version

for your Delegates?  Do you have HEIC installed and a current version?

 I really do not see any red stain in the places marked when I do it. Perhaps you do not have a current version of libheif

How are you viewing the result? What viewer?

I am using IM 7.1.0-38  with libheif 1.12.0 and they look the same to me.

Here is my converted image.

[converted.avif.zip](https://github.com/ImageMagick/ImageMagick/files/8941691/converted.avif.zip)

**Reply by ElieSangeunLee** [reply]
Thank you for the answer. I checked the imaged you converted and it was what I look for.

This is what I get from the command.
```
❯ magick -version
Version: ImageMagick 7.1.0-37 Q16-HDRI x86_64 20104 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenMP(4.5)
Delegates (built-in): bzlib fontconfig freetype gslib heic jng jp2 jpeg lcms lqr ltdl lzma openexr png ps raw tiff webp xml zlib
Compiler: gcc (4.2)
```

> Perhaps you do not have a current version of libheif

I tried to install libheif but homebrew said that I had alread had it.
```
❯ brew install libheif
...
Warning: libheif 1.12.0_1 is already installed and up-to-date.
To reinstall 1.12.0_1, run:
  brew reinstall libheif
```


> How are you viewing the result? What viewer?

For viewer, I just drag and drop the image to the Google Chrome Version 102.0.5005.115 (Official Build) (x86_64)


+) I tested again with the cropped version(attached in this discussion) and there was no stain either. I guess it happens with only the big original 1500*1500 but I'm afraid I cannot share the original image file because of its copyright.

**Comment by doggy8088** [comment]
@fmw42 
I'm using ImageMagick 7.1.0-57 that downloaded from [complete portable application on Linux](https://imagemagick.org/archive/binaries/magick) at [ImageMagick - Download](https://imagemagick.org/script/download.php) page.

```
$ magick --version
Version: ImageMagick 7.1.0-57 Q16-HDRI x86_64 d68553b17:20221230 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI OpenMP(3.1)
Delegates (built-in): bzlib djvu fontconfig freetype jbig jng jpeg lcms lqr lzma openexr png raqm tiff webp x xml zlib
Compiler: gcc (4.2)
```

I can't see HEIC been installed and I can't convert PNG to AVIF format. What can I do?

**Reply by fmw42** [reply]
The IM developers will need to comment about why they did not include libheif.  I am not a Linux user so cannot advise about how to install it.
